vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,35 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	if err := checkSpecArgSafety(string(at)); err != nil {
+		return nil, err
+	}
+
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("exec `git ls-files --with-tree %v` failed: %v", at, err)
+	}
+	out = bytes.TrimSpace(out)
```

Comment:
> Please explain why `bytes.TrimSpace` is particularly broad and imprecise for this situation, while not being so for 7 other similar usages in this file. Has it got to do with the specific output format of `git ls-files`?

No, it's not specific to `git ls-files`, this comment would apply to other places that use it in the same way.

By "too broad" I mean the `bytes.TrimSpace` operation performs more things than are strictly required. It trims spaces, tabs, newlines, and other characters from the beginning of the input, which is not necessary here. It also trims the same whitespace characters from the end of input, which again is more than strictly required, which is to trim the last single newline character.

Since this is already the style in other places, you don't need to address this as part of this PR, as ideally it can be a refactor that fixes all instances. It's also not critical as behavior remains largely the same for most inputs (unless the first file name begins with a whitespace character, then output is not the same).